src/charm.py

```diff
@@ -839,11 +847,17 @@ def _generate_tls_block(
             }
         }
 
-    def _generate_per_app_config(self, data: "RequirerData_IPA") -> Tuple[dict, str]:
-        prefix = self._get_prefix(data)
-
-        lb_servers = [{"url": f"http://{data['host']}:{data['port']}"}]
-        return self._generate_config_block(prefix, lb_servers, data)
+    def _generate_per_app_config(
+        self,
+        data: "IngressRequirerData",
+    ) -> Tuple[dict, str]:
+        prefix = self._get_prefix(data.app)
+        scheme = data.app["scheme"].value
```

Comment:
Are we sure that `scheme` is always in `data.app`? 🤔 
Shouldn't we default to `http` just in case?

Reply:
data is not `RelationData` but `IngressRequirerData` (our own type), and when unmarshalling the relation data we already default to http if no scheme is given. So at this point yes, it is guaranteed that data.app has 'scheme'.